apps/lib/cmp_mock_srv.c

```diff
@@ -325,6 +345,17 @@ static OSSL_CMP_ITAV *process_genm_itav(mock_srv_ctx *ctx, int req_nid,
                                                 ctx->newWithOld,
                                                 ctx->oldWithNew);
         break;
+    case NID_id_it_KemCiphertextInfo:
+        if (OSSL_CMP_ITAV_get0_value(req) == NULL) {
+            X509 *kemcert;
+
+            kemcert = extracert_withKEM(OSSL_CMP_MSG_get_extraCerts(genm));
```

Comment:
I agree it is not the best way and also certificate validation need to be performed ( will add TODO).
In case of client side having KEM cert, first genm message is without protection with extracert, So it does not populate validatedSevCert.
In case of server having KEM cert, it is assumed client already have valid server cert (in this implementation).